BENTO-1852, BENTO-1843

knockknockyoo opened the pull request:
Allowed User Login in DIfferent IDPS
Added NIH Login

n2iw commented:
- service/auth.js looks like it should be renamed to something like nih-auth.js
- routes/auth.js /login needs an additional parameter "IDP", and this parameter should be case-insensitive, meaning sending either "nih" and "NIH" should work
- controllers/auth-api.js: it seems nihLogout function is not used any where, if so, it should be removed. If nihLogout function is removed, then there is no need to separate this one function (logout) into a file, better just keep it in file routes/auth.js
-

knockknockyoo commented:
> Thank you for your review. 
Renamed nih-auth.js in service/auth.js.
Added missing parameters such as IDP, code, redirectURI
 - created a string utility to compare case-insensitive string including test codes
Removed nihLogout in controlllers/auth-api.js

n2iw reviewed (approved):
Looks good now!
